### ISSUE TITLE
decompressor_unexpected_chroma_sampling when transcoding image from CandyCam app.

### ISSUE THREAD

**Issue by MarsVard** [facebookincubator/spectrum#8]
I'm not sure if it's a specific issue with their app or if something is missing from this library. But I noticed this in my user base.

I will update this thread as soon as I have more information.

```
W/SpectrumLogger: requestid:5, error:decompressor_unexpected_chroma_sampling
    SpectrumException{name='decompressor_unexpected_chroma_sampling', message='decompressor_unexpected_chroma_sampling', location='image::ChromaSamplingMode facebook::spectrum::plugins::jpeg::LibJpegDecompressor::_chromaSamplingMode():103', description='decompressor_unexpected_chroma_sampling#-image::ChromaSamplingMode facebook::spectrum::plugins::jpeg::LibJpegDecompressor::_chromaSamplingMode():103'}
```

**Comment by lambdapioneer** [comment]
Hey @MarsVard, thanks for rising the issue!

I've been able to reproduce this by capturing an image with "Candy Camera" and decoding/transcoding with the sample Spectrum app. I've also downloaded the image to my development device and observed that it is using 4:1:1 sampling:

```
$ identify -verbose myimage.jpg | grep sampling
    jpeg:sampling-factor: 4x1,1x1,1x1
```

Our C++ code is incomplete regarding this particular sub-sampling variant of 4:1:1. It's also missing 4:4:0. The reason for this is that we didn't have evidence that these versions are practically used. This seems to be the case though and I'll try push an update this week,.

Code pointer of supported chroma sub-sampling to date: https://github.com/facebookincubator/spectrum/blob/master/cpp/spectrum/image/Specification.h#L23-L31 

```cpp
/**
 * Chroma sampling describe the number of luma information per chroma
 * information.
 */
enum class ChromaSamplingMode {
  S444 = 0,
  S420 = 1,
  S422 = 2,
};
```

This is a super helpful information for us, thanks a lot for sharing!

**Comment by lambdapioneer** [comment]
Fixed in master by https://github.com/facebookincubator/spectrum/commit/d36ba0e02c201acbf521841998401b710be3b7a7 and will be going out with the next release :)

**Comment by MarsVard** [comment]
That's fast! Awesome!

**Comment by lambdapioneer** [comment]
This is fixed with the 1.0.0 release: https://github.com/facebookincubator/spectrum/releases/tag/v1.0.0

Closing :)
